[CI] Trigger `uberjar` workflow manually

nemanjaglumac opened the pull request.

nemanjaglumac committed:
[CI] Trigger `uberjar` workflow manually

ariya commented on .github/workflows/uberjar.yml:
```diff
@@ -1,14 +1,11 @@
+    inputs:
+      commit:
+        description: 'A full-length commit SHA-1 hash'
+        required: true
```
Will a branch name also work here?

nemanjaglumac replied:
It might.
But one can select a branch before pasting the commit hash in the "Run workflow" manual trigger.

ariya replied:
If the branch name is given, would the image tagged with the branch name? This was useful back when this workflow is triggered automatically (as part of a PR).

nemanjaglumac replied:
Ok, I checked and regardless of whether you provide the commit SHA-1 or not, it will always print the branch name. For people who want always the latest commit in the given branch, this works perfectly.

However, if someone wants to build the specific commit, this might be inconvenient.
Imagine someone building commit `xyz` on `master` when commit `abc` on master already exists. The `xyz` image would overwrite it (which is basically the mechanism behind `metabase-head` and `metabase-enterprise-head` latest tag).

Speaking about, if we proceed with this change, both `metabase-head` and `metabase-enterprise-head` would stop updating for every commit to `master`. Since `metabase-head` has more than 100K pulls, we cannot just stop updating those images.

The first step is to remove a commit SHA-1 input as required and then to continue adding tweaks to this workflow to make it work as it did before for `master` branch. At the same time, we definitely want to move to the manual trigger for `metabase-dev` (builds on PRs).

nemanjaglumac committed:
Do not make commit SHA input required

nemanjaglumac committed:
Continue building on push to `master`

This will keep `metabase-head` and `metabase-enterprise-head`
DockerHub repos working as they did before.

nemanjaglumac committed:
Simplify by removing references to `main` branch

nemanjaglumac committed:
Ignore unrelated paths for push on `master` branch
